src/farcasterd/trade_state_machine.rs

```diff
@@ -173,6 +178,20 @@ pub struct SwapdRunning {
     trade_role: TradeRole,
 }
 
+pub enum ConsumedOfferRole {
+    Taker,
+    Maker(Commit),
+}
+
+impl ConsumedOfferRole {
+    pub fn trade_role(&self) -> TradeRole {
```

Comment:
1a2b9097e459b4e238fa5323ce688091feef5cc6

Reply:
...not sure it was a good indication from me as `From` consumes the value et we'd like to avoid that. This is minor optimization tho, I'm fine keeping the patch or removing it. Sorry for the noise